File: packages/api/utils/alert.js
const { WebClient } = require('@slack/web-api');
const config = require('./config');

/**
 * @description sends alert
 * https://www.npmjs.com/package/@slack/web-api
 */
const sendAlert = async ({ channel=config.slack.alertChannel, text }, {client = new WebClient(config.slack.token)} = {}) => {
  try {
    const data = await client.chat.postMessage({
      text,
      channel,
    })
    
    return { data }
  } catch (error) {
    return { error }
  }
};



module.exports = {
    sendAlert
}

File: packages/airflow/dags/optimise_keywords/utils/alert.py
import logging, os
# Import WebClient from Python SDK (github.com/slackapi/python-slack-sdk)
from slack_sdk import WebClient
from slack_sdk.errors import SlackApiError
from config import config

# WebClient instantiates a client that can call API methods
# When using Bolt, you can use either `app.client` or the `client` passed to listeners.

def alert(message):
    try:

        client = WebClient(token=config["slack"]["token"])
        
        # Call the conversations.list method using the WebClient
        client.chat_postMessage(
            channel=config["slack"]["alert_channel"],
            text=message
        )
   
    except SlackApiError as e:
        print(f"Error: {e}")